src/Microsoft.AspNet.Mvc.Core/OptionDescriptors/OptionDescriptorExtensions.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Collections.Generic;
+using Microsoft.AspNet.Mvc.OptionDescriptors;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Extension methods for collections of option descriptors.
+    /// </summary>
+    public static class OptionDescriptorExtensions
+    {
+        /// <summary>
+        /// Returns the only instance of <typeparamref name="TInstance"/> from a sequence of option descriptors.
```

Comment:
InstanceOf() is supposed to throw, and InstanceOfOrDefault is that one that returns null